static/js/util/profile_edit.js

```diff
@@ -47,14 +49,35 @@ function boundSelectField(key, label, options, error) {
   </div>;
 }
 
+// bind this to this.boundSelectField in the constructor of a form component
+// to update select fields
+// pass in the name (used as placeholder), key for profile, and the options.
+export function boundDateField(key, label, error) {
+  const { updateProfile, profile } = this.props;
+  let onChange = date => {
+    let clone = Object.assign({}, profile);
+    // format as ISO-8601
+    clone[key] = date.format("YYYY-MM-DD");
+    updateProfile(clone);
+  };
+  let newDate = profile[key] ? moment(profile[key], "YYYY-MM-DD") : null;
+  return <div>
+    <DatePicker
+      selected={newDate}
+      placeholderText={label}
+      showYearDropdown
+      onChange={onChange}
+    />
+    <span className="validation-error-text">{error}</span>
+  </div>;
+}
```

Comment:
this is nice and clean! I like the way we've set up these functions, it's a nice pattern that encourages lots of re-use.